### ISSUE TITLE
🐛 Bug: Rights issues for rpm installation on Redhat Linux

### ISSUE THREAD

**Issue by quentin-rey** [svalabs/sva.sentinelone#30]
##### SUMMARY

Rights issues for rpm installation

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

`sva.sentinelone.install_agent : Linux: Install unsigned .rpm agent package via rpm`

##### ANSIBLE VERSION

```
ansible [core 2.16.4]
  config file = [MY_CFG_PATH]/ansible.cfg
  configured module search path = ['/home/ansible/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/ansible/.local/lib/python3.11/site-packages/ansible
  ansible collection location = /home/ansible/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/ansible/.local/bin/ansible
  python version = 3.11.5 (main, Sep 22 2023, 15:34:29) [GCC 8.5.0 20210514 (Red Hat 8.5.0-20)] (/usr/bin/python)
  jinja version = 3.1.3
  libyaml = True
```

##### COLLECTION VERSION

```
# /home/ansible/.ansible/collections/ansible_collections
Collection      Version
--------------- -------
sva.sentinelone 1.1.0
```

##### DEPENDENCY VERSIONS

```
ansible==9.3.0
ansible-core==2.16.4
argcomplete==3.2.2
bcrypt==4.1.2
certifi==2024.2.2
cffi==1.16.0
charset-normalizer==3.3.2
cryptography==42.0.5
deepdiff==6.7.1
docopt==0.6.2
idna==3.6
Jinja2==3.1.3
lxml==5.1.0
MarkupSafe==2.1.5
nsx-policy-python-sdk @ file://localhost//tmp/pip-req-build-fzzof76l/lib/nsx-policy-python-sdk/nsx_policy_python_sdk-4.1.2.0.0-py2.py3-none-any.whl#sha256=d8b09fddd3bfac077cc2c4484f764d1b364ece202e1d3575da3adbd00707930a
nsx-python-sdk @ file://localhost//tmp/pip-req-build-fzzof76l/lib/nsx-python-sdk/nsx_python_sdk-4.1.2.0.0-py2.py3-none-any.whl#sha256=cd5f5a47d2bf066f047d5eba2d6bbd1f784fb36d7e1f734fa6adaab4d6d93fea
nsx-vmc-aws-integration-python-sdk @ file://localhost//tmp/pip-req-build-fzzof76l/lib/nsx-vmc-aws-integration-python-sdk/nsx_vmc_aws_integration_python_sdk-4.1.2.0.0-py2.py3-none-any.whl#sha256=d141ddcbb4dd7500299070129ebed10cdb5a3bf15c1b48de791760e81fbb27a7
nsx-vmc-policy-python-sdk @ file://localhost//tmp/pip-req-build-fzzof76l/lib/nsx-vmc-policy-python-sdk/nsx_vmc_policy_python_sdk-4.1.2.0.0-py2.py3-none-any.whl#sha256=d20503b7214d578bc996d33c15385e980f4ee8b381b0615e8ee9c9693ff929ae
ordered-set==4.1.0
packaging==23.2
paramiko==3.4.0
prompt-toolkit==3.0.43
pycparser==2.21
pyflakes==3.2.0
Pygments==2.17.2
pykerberos==1.2.4
PyNaCl==1.5.0
pyOpenSSL==24.0.0
pyspnego==0.10.2
pyvim==3.0.3
pyvmomi==8.0.2.0.1
pywinrm==0.4.3
PyYAML==6.0.1
requests==2.31.0
requests-ntlm==1.2.0
resolvelib==1.0.1
six==1.16.0
urllib3==2.2.1
vapi-common-client @ file://localhost//tmp/pip-req-build-fzzof76l/lib/vapi-common-client/vapi_common_client-2.44.0-py2.py3-none-any.whl#sha256=b4f906ebc17406af019a2e03a675f7404a6ad08c10b3bba99d493c30929b667f
vapi-runtime @ file://localhost//tmp/pip-req-build-fzzof76l/lib/vapi-runtime/vapi_runtime-2.44.0-py2.py3-none-any.whl#sha256=71eafdaa9123f87b1bfb27e6903876e8ce192b9d80248135532577a6442477ad
vcenter-bindings @ file://localhost//tmp/pip-req-build-fzzof76l/lib/vcenter-bindings/vcenter_bindings-4.2.0-py2.py3-none-any.whl#sha256=055cc278d9f9fdf0ae61c9f009aaabaa8696dd13b89c48061c637f4bf3105cbf
vmwarecloud-aws @ file://localhost//tmp/pip-req-build-fzzof76l/lib/vmwarecloud-aws/vmwarecloud_aws-1.64.0-py2.py3-none-any.whl#sha256=2a57ce72812a355c4bbd5a5d0bfff04617d85402da96911d9dae71397d6529a9
vmwarecloud-draas @ file://localhost//tmp/pip-req-build-fzzof76l/lib/vmwarecloud-draas/vmwarecloud_draas-1.23.0-py2.py3-none-any.whl#sha256=629cae86d5bcea0317242135d06e03a8f96707ae3913770745ef5e728e9c6b38
vsphere-automation-sdk @ git+https://github.com/vmware/vsphere-automation-sdk-python.git@99ee630cb819495f572e9775b48ce0ac1b0263de
wcwidth==0.2.13
xmltodict==0.13.0
```

##### CONFIGURATION

```
CONFIG_FILE() = [MY_CFG_PATH]/ansible.cfg
DEFAULT_FORKS([MY_CFG_PATH]/ansible.cfg) = 50
DEFAULT_PRIVATE_KEY_FILE([MY_CFG_PATH]ansible.cfg) = /home/ansible/.ssh/ansible
DEFAULT_REMOTE_USER([MY_CFG_PATH]/ansible.cfg) = MY_REMOTE_USERNAME
HOST_KEY_CHECKING([MY_CFG_PATH]/ansible.cfg) = False
INVENTORY_ENABLED([MY_CFG_PATH]/ansible.cfg) = ['vmware_vm_inventory', 'yaml', 'ini']
```

##### OS / ENVIRONMENT

```
NAME="Red Hat Enterprise Linux"
VERSION="8.9 (Ootpa)"
ID="rhel"
ID_LIKE="fedora"
VERSION_ID="8.9"
PLATFORM_ID="platform:el8"
PRETTY_NAME="Red Hat Enterprise Linux 8.9 (Ootpa)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:redhat:enterprise_linux:8::baseos"
HOME_URL="https://www.redhat.com/"
DOCUMENTATION_URL="https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8"
BUG_REPORT_URL="https://bugzilla.redhat.com/"

REDHAT_BUGZILLA_PRODUCT="Red Hat Enterprise Linux 8"
REDHAT_BUGZILLA_PRODUCT_VERSION=8.9
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="8.9"
```

##### STEPS TO REPRODUCE

Launch a playbook using the agent deployment role.

```bash
ansible-playbook -i [MY_INVENTORY_PATH] -e "[MY_EXTRA_VARS]" install_sentinelone_agent.yml -vvv
```

`install_sentinelone_agent.yml`:

```
- name: Sentinelone Agent Deployment
  hosts: "{{ target_hosts }}"
  gather_facts: true
  tasks:
    - name: "Install specific agent version on Linux"
      ansible.builtin.include_role:
        name: sva.sentinelone.install_agent
      vars:
        console_url: "MY_URL"
        api_token: "MY_TOKEN"
        site: "Default site"
        group: "MY_GROUP"
        gpg_key: "sentinel_one.gpg"
        agent_version: 'custom'
        custom_version: '23.2.2.4'
        hide_sensitive: false
```

##### EXPECTED RESULTS

The role should be able to perform RPM tasks.

##### ACTUAL RESULTS

Currently, the task is in error because it is not being run in sudo.

```yaml
error: can't create transaction lock on /var/lib/rpm/.rpm.lock (Permission denied)
```

In my case, I use a `remote_user`, who has sudoers rights on my remote hosts. Without the presence of `become: true` on the RPM tasks, this won't work.

----

It seems to me that the solution would be to add `become: true` to the RPM tasks.

For example:
```yaml
- name: "Linux: Install unsigned .rpm agent package via rpm"
  ansible.builtin.command:
    cmd: "rpm -i --nodigest {{ remote_pkg_path }}"
    creates: "/opt/sentinelone/bin/sentinelctl"
  when: not signed_package
```
should be
```yaml
- name: "Linux: Install unsigned .rpm agent package via rpm"
  become: true
  ansible.builtin.command:
    cmd: "rpm -i --nodigest {{ remote_pkg_path }}"
    creates: "/opt/sentinelone/bin/sentinelctl"
  when: not signed_package
```

**Comment by mwester117** [comment]
Hi @quentin-rey,
thanks for your report. Indeed I tested the role with `become: true` for global scope. As a workaround I recommend adding `become: true` to your vars for the _include_role_ task. I will update the according tasks in the role soon.

**Comment by quentin-rey** [comment]
I tried this during my debug, but the problem is that I'm launching my playbook with a local user who isn't sudoer and isn't the one I'm using remotely (which has sudoer rights).

----

Potentially, the problem could also arise for installation on DEB based systems.

How do you plan to integrate `become: true`? At task level or at block level?

**Comment by mwester117** [comment]
@quentin-rey: Finally this issue is fixed in collection release v1.1.1.
The release is now published:
https://github.com/svalabs/sva.sentinelone/releases/tag/v1.1.1
https://galaxy.ansible.com/ui/repo/published/sva/sentinelone/
